### PR TITLE
ci: use amplify windows worker for smoke tests

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes

This PR:
- uses `amplify-cli_windows-latest_8-core` as window worker, our e2e test framework demands bigger machine size.
- removes "simplified windows smoke tests" and replaces them with `yarn smoketest` (this is already set up just requires logical conditions tuning)
- adds couple of changes in e2e tests to make it work, see comments in the code.

#### Description of how you validated changes

By running tests:
See passing job https://github.com/aws-amplify/amplify-cli/actions/runs/6086210651

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/actions/run-smoke-tests/action.yml [hunk 1]
```diff
@@ -21,6 +21,14 @@ runs:
         echo "AMPLIFY_PATH=$AMPLIFY_PATH"
         $AMPLIFY_PATH version
 
+    - name: Remove Pkg Fetch Checksum on Windows
+      shell: bash
+      if: runner.os == 'Windows'
+      run: |
+        cd scripts
+        yarn
+        yarn ts-node remove-pkg-fetch-checksum.ts
```
Comment (sobolk) [inline .github/actions/run-smoke-tests/action.yml:30]: See:
https://github.com/aws-amplify/amplify-cli/blob/fe043d5e222bf93e19f3923362c4beff6677609c/codebuild_specs/build_windows.yml#L9

File: .github/workflows/smoke-tests-manual.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ on:
       os:
         required: false
         type: string
-        default: '["macos-latest-xl", "ubuntu-latest", "windows-latest"]'
+        default: '["macos-latest-xl", "ubuntu-latest", "amplify-cli_windows-latest_8-core"]'
```
Comment (sobolk) [inline .github/workflows/smoke-tests-manual.yml:14]: I have attempted to make this list a bit smoother i.e. `macos,ubuntu,windows`. However, it makes actions quite complicated, so I parked that effort.
End of the day, when we run this manually, this parameter is populated with this default, all operator has to do is to remove values.

**Review by 0618 — APPROVED** [review]

File: .github/workflows/smoke-tests-manual.yml (hunk 2 above)
Comment (0618) [reply .github/workflows/smoke-tests-manual.yml:14]: Where is the `amplify-cli_windows-latest_8-core` from?

Comment (sobolk) [reply .github/workflows/smoke-tests-manual.yml:14]: https://t.corp.amazon.com/P98348294/communication

Comment (sobolk) [reply .github/workflows/smoke-tests-manual.yml:14]: Also a visual.
image

**Review by edwardfoyle — APPROVED** [review]

**Pull request merged**